skills/repo.go

```diff
@@ -19,10 +19,200 @@
 
 package skills
 
+import (
+	"errors"
+	"sort"
+	"sync"
+)
+
 // Repository stores all known snappy skills and slots and types.
-type Repository struct{}
+type Repository struct {
+	// Protects the internals from concurrent access.
+	m      sync.Mutex
+	types  []Type
+	skills []*Skill
+}
+
+var (
+	// ErrTypeNotFound is reported when skill type cannot found.
+	ErrTypeNotFound = errors.New("skill type not found")
+	// ErrDuplicateType is reported when type with duplicate name is being added to a repository.
+	ErrDuplicateType = errors.New("duplicate type name")
+	// ErrSkillNotFound is reported when skill cannot be looked up.
+	ErrSkillNotFound = errors.New("skill not found")
+	// ErrDuplicateSkill is reported when skill with duplicate name is being added to a repository.
+	ErrDuplicateSkill = errors.New("duplicate skill name")
```

Comment:
Drop all of these please. Replace them by inline fmt.Errorf with proper information reported.